### DISCUSSION TITLE
Remote Zellij doesn't see session tabs

### DISCUSSION THREAD

**Discussion by abitrolly** [zellij-org/zellij#3700] [Q&A] [answered]
I've got Zellij session open in Linux console `tty1` (no GUI, X11, etc), and I want to copy its tab content from remote machine.

When I connect from remote machine over SSH, `zellij ls` shows the session as EXITED.

```sh
% zellij ls
erudite-pepper [Created 9h 49m 9s ago] (EXITED - attach to resurrect)
```

But when I attach to it with `zellij attach erudite-pepper`, only the new blank tab is shown.
The session in `tty1` in Linux console is still active - I can run commands there and scroll back.
But why `zellij` connected from SSH doesn't show its contents?

```sh
% zellij --version
zellij 0.40.1
```

**Comment by imsnif** [comment, marked as answer]
Just a guess, but you might want to make sure `$XDG_RUNTIME_DIR` is identical in both instances.

**Reply by abitrolly** [reply]
Interesting.

For SSH session, `$XDG_RUNTIME_DIR` is empty (I am using Dropbear SSH server).
For `tty1`, the `$XDG_RUNTIME_DIR` is set to `/run/user/1000`.

But there is no desktop environment on this box. Is there any other place to securely share user data between sessions?

**Reply by abitrolly** [reply]
That was the right guess. When I run `export XDG_RUNTIME_DIR=/run/user/1000` manually in SSH session, then sessions can be accessed.

**Comment by lgarron** [comment]
I commented in https://github.com/zellij-org/zellij/issues/3708#issuecomment-2522707489 as well, but I'm also seeing this issue.

Setting `$XDG_RUNTIME_DIR` did nothing — I can confirm the value is exactly the same in all environments, both where `zellij` is invoked and inside `zellij` sessions themselves.

**Reply by lgarron** [reply]
Oh! I tried `pkill zellij` on a whim, and that worked. Both the issue and the fix are… very not obvious. Is there documentation on why I would ever expect the default behaviour in the first place?

I feel like:

- Connecting over SSH is a use case `zellij` is very suited for.
- Even as a die-hard XDG spec user and advocate, I didn't have `$XDG_RUNTIME_DIR` set. 

I also have yet to find information about this on the [FAQ](https://zellij.dev/documentation/faq) or anywhere else in the docs. 😔
